app/uk/gov/hmrc/leakdetection/services/LeaksService.scala

```diff
@@ -63,13 +69,42 @@ class LeaksService @Inject()(ruleService: RuleService,
   }
 
   private def getTeamRepos(teamName: Option[String]): Future[Option[Seq[String]]] = teamName match {
-    case Some(t) => teamsAndRepositoriesConnector.team(t).map(_.map(t => t.repos.map(_.values.toSeq.flatten).toSeq.flatten))
+    case Some(t) => teamsAndRepositoriesConnector.team(t).map(_.map(_.repos.map(_.values.toSeq.flatten).toSeq.flatten))
     case None    => Future.successful(None)
   }
 
   private def filterLeaksByTeam(leaks: Seq[Leak], repos: Option[Seq[String]]): Seq[Leak] = repos match {
     case Some(r) => leaks.filter(l => r.contains(l.repoName))
-    case None => leaks
+    case None    => leaks
   }
 
+  def clearLeaksAfterBranchDeleted(deleteBranchEvent: DeleteBranchEvent): Future[Long] =
+    leakRepository.removeBranch(deleteBranchEvent.repositoryName, deleteBranchEvent.branchRef)
+
+  def saveLeaks(repo:Repository, branch:Branch, leaks: Seq[Leak]):Future[Unit] =
+    leakRepository.update(repo.asString, branch.asString, leaks).map(_ => ())
+
+  override def metrics(implicit ec: ExecutionContext): Future[Map[String, Int]] =
+    for {
+      total      <- leakRepository.countAll()
```

Comment:
Should this not be the total number of reports?

Reply:
yes, but its a bit of a pain to get from the leaks-repository and ultimately it isn't a very useful/meaningful metric